### PR TITLE
Fix: Fixed OverflowException in UpdateSelectionSize

### LINKED ISSUE #15500: Bug: BaseLayoutPage.UpdateSelectionSize OverflowException
### Description

**BaseLayoutPage.UpdateSelectionSize ()**
System.OverflowException: Arithmetic operation resulted in an overflow.

### Steps To Reproduce

Unknown

### Requirements

- Resolve crash

### Files Version

v3.4.4

### Windows Version

Windows 11

### Log File

```csharp
System.Linq
Enumerable.Sum[TSource,TResult,TAccumulator] (IEnumerable`1 source, Func`2 selector)
Files.App.Views.Layouts
BaseLayoutPage.UpdateSelectionSize ()
Files.App.Views.Shells
BaseShellPage.FilesystemViewModel_DirectoryInfoUpdated (Object sender, EventArgs e)
System.Threading.Tasks.Task
<>c.<ThrowAsync>b__128_0 (Object state)
WinRT
ExceptionHelpers.<ThrowExceptionForHR>g__Throw|39_0 (Int32 hr)
ABI.Windows.ApplicationModel.Core
IUnhandledErrorMethods.Propagate (IObjectReference _obj)
Microsoft.AppCenter.Utils
ApplicationLifecycleHelperWinUI.<ctor>b__0_3 (Object sender, UnhandledErrorDetectedEventArgs eventArgs)
```

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes #15500

**Steps used to test these changes**

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Opened Files ...
2. ...

### CODE REVIEW

File: src/Files.App/Views/Layouts/BaseLayoutPage.cs [hunk 1]
```diff
@@ -702,8 +702,8 @@ public void UpdateSelectionSize()
 			var isSizeKnown = !items.Any(item => string.IsNullOrEmpty(item.FileSize));
 			if (isSizeKnown)
 			{
-				long size = items.Sum(item => item.FileSizeBytes);
-				SelectedItemsPropertiesViewModel.ItemSizeBytes = size;
+				ulong size = items.Aggregate(0UL, (current, item) => current + (ulong)item.FileSizeBytes);
+				SelectedItemsPropertiesViewModel.ItemSizeBytes = (long)size;
```
Comment (0x5bfa) [inline src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]: Is casting here ok when overflow exception is happening?

Comment (0x5bfa) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]: I'd suggest 
```suggestion
				SelectedItemsPropertiesViewModel.ItemSizeBytes = size >= (ulong)long.MaxValue ? -1l : (long)size; // NOTE: Suppressed OverflowException in #16165
```
What do you think?

Comment (yaira2) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]: Fyi @hishitetsu

Comment (hishitetsu) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]: OverflowException may occur even in ulong.
How about using decimal?

Comment (yaira2) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]: > OverflowException may occur even in ulong.

True, but it's already very uncommon with long (2 reports) and switching to ulong should take care of it.

Comment (yaira2) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]: Does `ToSizeString` work with decimal?

Comment (0x5bfa) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]: Hmm, in the first place WIN32_FIND_DATA, which we use for enumeration of items, can only retrieve up to sizeof(ulong)

Comment (0x5bfa) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]: Compromise may be necessary.

Btw, should also be updated to ulong.
https://github.com/files-community/Files/blob/8ba4f4517a35e85547a89ec63a08805e00f60965/src/Files.App/Data/Items/ListedItem.cs#L281

Comment (hishitetsu) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]: > Does `ToSizeString` work with decimal?

We can cast decimal to double.

Comment (hishitetsu) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]: Casting decimal to long may throw an OverflowException.

Comment (0x5bfa) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs]: What's the reason to do so?
If I understand C#'s value types correctly, ulong can express larger value (20 digits) than double (16 digits). decimal can express much larger (like 29?) than ulong or double though.

Comment (0x5bfa) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs]: You are right regarding the first part. But isn't the significant number the same even when it comes to integers? Like 1.00 == 100

The max value that can be expressed accurately (according to an article):
Type|Max integer
---|--:
double|`0x20000000000000`
ulong|`0xFFFFFFFFFFFFFFFF`

double.MaxValue is 0x7FEFFFFFFFFFFFFF so even this is less than ulong.MaxValue...

But you seem to understand much better, I'm fine to merge.

Comment (0x5bfa) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs]: You are right 😄
How large is that digits?

Comment (hishitetsu) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs]: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/floating-point-numeric-types

> ±5.0 × 10^−324 to ±1.7 × 10^308

Comment (yaira2) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs]: @hishitetsu can you check my latest commit?

Comment (hishitetsu) [reply src/Files.App/Views/Layouts/BaseLayoutPage.cs]: Actually, I haven't been able to check it out yet. I think I will be able to see it tomorrow morning in Japan time.

File: src/Files.App/Data/Models/SelectedItemsPropertiesViewModel.cs [hunk 2]
```diff
@@ -183,8 +183,8 @@ public bool IsUncompressedItemSizeVisibile
 			set => SetProperty(ref isUncompressedItemSizeVisibile, value);
 		}
 
-		private long itemSizeBytes;
-		public long ItemSizeBytes
+		private double itemSizeBytes;
+		public double ItemSizeBytes
```
Comment (hishitetsu) [inline src/Files.App/Data/Models/SelectedItemsPropertiesViewModel.cs:187]:
```suggestion
		private decimal itemSizeBytes;
		public decimal ItemSizeBytes
```

File: src/Files.App/Views/Layouts/BaseLayoutPage.cs [hunk 3]
```diff
@@ -702,8 +702,8 @@ public void UpdateSelectionSize()
 			var isSizeKnown = !items.Any(item => string.IsNullOrEmpty(item.FileSize));
 			if (isSizeKnown)
 			{
-				long size = items.Sum(item => item.FileSizeBytes);
-				SelectedItemsPropertiesViewModel.ItemSizeBytes = size;
+				decimal size = items.Sum(item => item.FileSizeBytes);
+				SelectedItemsPropertiesViewModel.ItemSizeBytes = (double)size;
```
Comment (hishitetsu) [inline src/Files.App/Views/Layouts/BaseLayoutPage.cs:706]:
```suggestion
				SelectedItemsPropertiesViewModel.ItemSizeBytes = size;
```

**Review by hishitetsu — COMMENTED** [review]
I think it is better to unify with decimal.

**Review by hishitetsu — APPROVED** [review]

**Pull request merged**